packages/data-context/src/sources/ProjectDataSource.ts

```diff
@@ -252,20 +252,26 @@ export class ProjectDataSource {
       throw new Error('Cannot start spec watcher without current project')
     }
 
-    const onSpecsChanged = debounce(async () => {
+    // When file system changes are detected, we retrieve any spec files matching
+    // the determined specPattern. This function is debounced to limit execution
+    // during sequential file operations.
+    const onProjectFileSystemChange = debounce(async () => {
       const specs = await this.findSpecs(projectRoot, testingType, specPattern, excludeSpecPattern, additionalIgnore)
 
       this.ctx.actions.project.setSpecs(specs)
-    })
+    }, 200)
 
-    this._specWatcher = chokidar.watch(specPattern, {
+    // We respond to all changes to the project's filesystem when
+    // files or directories are added and removed that are not explicitly
+    // ignored by config
+    this._specWatcher = chokidar.watch('.', {
```

Comment:
So are we saying that providing the `specPattern` here definitely does not work properly?

Reply:
From my testing, yes, it does not work correctly when files are introduced with or removed alongside parent directories. No 'add'/'unlink' events for those files get emitted in that case. This could very well be a bug within chokidar, I need to dig into its implementation more fully to figure out why it's behaving this way.

Reply:
Interesting, it looks like chokidar v4 is dropping globbing entirely: https://github.com/paulmillr/chokidar/pull/1195, so this might be the best approach, to watch the directory and handle matching on our side.